amoeba.h

```diff
@@ -517,6 +517,7 @@ AM_API void am_delvariable(am_Var *var) {
     if (var && --var->refcount <= 0) {
         am_Solver *solver = var->solver;
         am_VarEntry *e;
+        if (solver->dirty_vars.id != 0) am_updatevars(solver);
```

Comment:
you should use `if (var->dirty_next.type == AM_DUMMY)` to check whether the var is dirty.